fake_spectra/singleabs.h

```diff
@@ -56,10 +56,17 @@ class SingleAbsorber
          * aa: voigt_fac/btherm the parameter for the Voigt profile
          * Check extra carefully whether m_vhigh is real.
          */
-        SingleAbsorber(double bth_i, double vdr2_i, double vsm_i, double aa_i):
+        SingleAbsorber(double bth_i, double vdr2_i, double vsm_i, double aa_i, int arepo_):
             btherm(bth_i), vdr2(vdr2_i), vsmooth(vsm_i), aa(aa_i),
-            m_vhigh((vsmooth*vsmooth > vdr2 ? sqrt(vsmooth*vsmooth-vdr2) : 0))
-        {};
+            m_vhigh((vsmooth*vsmooth > vdr2 ? sqrt(vsmooth*vsmooth-vdr2) : 0)),
+            arepo(arepo_)
+            {
+                if(arepo)
+                {
+                    if(vdr2 > 0 && vsmooth > 0) m_vhigh = (vsmooth - vdr2)/2.;
+                    else m_vhigh = 0;
```

Comment:
Please see the next comment. When arepo == True, m_vhigh = (arr2[2*ind+1] - arr2[2*ind])/2. I changed the meaning of dr2 and vsmooth in this case (because I wanted to keep the original structure of the code...)